### PR TITLE
Rate Limiting

### PR DESCRIPTION
**Pull request by versx**
Fix: #146 
Removes csrf token package, adds rate limiting package.
- Defaults are set to 6000 per hour (100 requests per minute) although configurable.

TODO: Limit queried rows to 1000/10000

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -139,6 +134,8 @@ app.use(async (req, res, next) => {
     res.redirect('/login');
 });
 
+app.use('/', requestRateLimiter);
```
Comment (Mygod) [inline src/index.js:137]: I think you only want to apply it on `/api`.

File: src/views/index-js.mustache [hunk 2]
```diff
@@ -1572,8 +1572,7 @@ function loadSearchData (id, value) {
             value: value,
             lat: center.lat,
             lon: center.lng,
-            icon_style: selectedIconStyle,
-            _csrf: '{{csrf}}'
+            icon_style: selectedIconStyle
```
Comment (Mygod) [inline src/views/index-js.mustache:1575]: A trailing comma is valid js and is more git friendly.

**Review by Mygod — APPROVED** [review]
Seems like a much better fix of #146 :+1:

**Comment by Mygod** [comment]
Also it is better to give the user some visual clue as to when rate limit applied. This would help debugging for map admins as well as dumb users.

**Comment by versx** [comment]
> Also it is better to give the user some visual clue as to when rate limit applies. This would help debugging for map admins as well as dumb users.

If they reload they are greeted with the rate limited message. Although maybe redirecting to a 429.mustache page would be better/more elegant.

**Comment by Mygod** [comment]
Even after only applying rate limit on only api paths?

**Comment by versx** [comment]
> Even after only applying rate limit on only api paths?

Good point, I did not test after changing to only api endpoints being affected.

**Comment by Mygod** [comment]
So you merged this without adding rate limit notifications.

**Comment by versx** [comment]
> So you merged this without adding rate limit notifications.

https://github.com/WatWowMap/MapJS/pull/150/files#diff-1fdf421c05c1140f6d71444ea2b27638R32
It's there, it just doesn't actually redirect for some reason.

**Comment by Mygod** [comment]
facepalm

Client issues API requests using AJAX. Redirecting those requests won't show up in the actual map.

**Comment by versx** [comment]
It's not triggered from the ajax requests, it's triggered the first time the rate limit is hit. It's what it mentions in the docs, it fires off that method but it doesn't redirect the client.

**Comment by Mygod** [comment]
But you only apply rate limiting on `/api` which is only used via AJAX.

**Comment by versx** [comment]
image

**Comment by Mygod** [comment]
Yes. That means that you need to write a separate UI for rate limiting.

**Comment by versx** [comment]
That's what this route is for, it's in the UI not API, so it's not rate limited
https://github.com/WatWowMap/MapJS/pull/150/files#diff-72d4dc7f57ee28cd5dde9a2070d6840eR76

**Pull request merged**